### PR TITLE
[876] Add tests for Tokens Studio sync

### PR DESCRIPTION
**Pull request by georgebuciuman**
### Why does this PR exist?

Closes https://github.com/tokens-studio/studio-app/issues/876

### What does this pull request do?

- Adds tests for all Tokens Studio sync flows
- Fixes a bug when moving a theme to a new group
- Removes logic to resolve dynamic sets since we get the tokens from BE now

### CODE REVIEW

**Review by PiotrKalinski — APPROVED** [review]

File: packages/tokens-studio-for-figma/src/app/store/models/reducers/__tests__/tokensStudio.test.ts [hunk 1]
```diff
@@ -0,0 +1,864 @@
...
+    store = init<RootModel>(storeInitialState);
+
+    graphqlExecSpy = jest.spyOn(Graphql, 'exec');
+    graphqlOpSpy = jest.spyOn(Graphql, 'op');
+  });
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it.each([
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/app/store/models/reducers/__tests__/tokensStudio.test.ts:164]: TIL!

File: packages/tokens-studio-for-figma/src/storage/TokensStudioTokenStorage.ts [hunk 2]
```diff
@@ -73,32 +72,7 @@ async function getProjectData(urn: string): Promise<ProjectData | null> {
...
-    const dynamicTokenSets = tokenSets.filter((tokenSet) => tokenSet.type === 'DYNAMIC');
-    const dynamicTokenSetData = await Promise.all(
-      dynamicTokenSets.map(async (tokenSet) => {
-        if (!tokenSet.generatorUrn) {
-          return null;
-        }
-        const tokens = await fetchDynamicTokenSetData(tokenSet.generatorUrn);
-        return { ...tokenSet, tokens };
-      }),
-    );
```
Comment (six7) [inline packages/tokens-studio-for-figma/src/storage/TokensStudioTokenStorage.ts:87]: just out of interest, is this no longer needed? or was this never needed?

**Review by six7 — APPROVED** [review]
just a clarifying question, otherwise feel free to merge!

File: packages/tokens-studio-for-figma/src/storage/TokensStudioTokenStorage.ts (hunk 2 above)
Comment (georgebuciuman) [reply packages/tokens-studio-for-figma/src/storage/TokensStudioTokenStorage.ts:87]: it's no longer needed as Pop implemented resolving on backend. It was needed before that

**Pull request merged**